Author: Julian Andres Klode <jak@debian.org>
Description: Add support for GNU/kFreeBSD platforms to kbuild.
 It still has a few problems, like the missing fflagstostr() and strtofflags() functions,
 which have all been worked around for now.
Bug-Debian: http://bugs.debian.org/540665

--- a/src/kmk/incdep.c
+++ b/src/kmk/incdep.c
@@ -678,7 +678,7 @@ incdep_are_threads_enabled (void)
   if (getenv ("KMK_THREADS_ENABLED"))
     return 1;
 
-#if defined (__gnu_linux__) || defined (__linux__)
+#if defined (__gnu_linux__) || defined (__linux__) || defined(__GLIBC__)
   /* Try detect fakeroot. */
   if (getenv ("FAKEROOTKEY")
    || getenv ("FAKEROOTUID")
--- a/kBuild/env.sh
+++ b/kBuild/env.sh
@@ -264,6 +264,10 @@ if test -z "$KBUILD_HOST"; then
             KBUILD_HOST=freebsd
             ;;
 
+        GNU/kFreeBSD)
+            KBUILD_HOST=gnukfbsd
+            ;;
+
         Haiku)
             KBUILD_HOST=haiku
             ;;
--- a/src/kmk/kmkbuiltin/install.c
+++ b/src/kmk/kmkbuiltin/install.c
@@ -209,7 +209,8 @@ kmk_builtin_install(int argc, char *argv
 			dodir = 1;
 			break;
 		case 'f':
-#ifdef UF_IMMUTABLE
+#if defined(UF_IMMUTABLE) && !defined(__GLIBC__)
+			/** @todo: we need strtofflags() exported in e.g. libbsd */
 			flags = optarg;
 			if (strtofflags(&flags, &fset, NULL))
 				return errx(EX_USAGE, "%s: invalid flag", flags);
--- a/src/kmk/kmkbuiltin/rm.c
+++ b/src/kmk/kmkbuiltin/rm.c
@@ -675,7 +675,8 @@ check(char *path, char *name, struct sta
                     )
 			return (1);
 		bsd_strmode(sp->st_mode, modep);
-#ifdef SF_APPEND
+#if defined(SF_APPEND) && !defined(__GLIBC__)
+		/** @todo: we need fflagstostr() exported in e.g. libbsd */
 		if ((flagsp = fflagstostr(sp->st_flags)) == NULL)
 			exit(err(1, "fflagstostr"));
 		(void)fprintf(stderr, "override %s%s%s/%s %s%sfor %s? ",
